### PR TITLE
Enable Embroider by default for new projects

### PR DESCRIPTION
**Pull request by thoov**
New apps generated via `ember new` will now default to use Embroider's build system.

### CODE REVIEW

**Review by rwjblue — APPROVED** [review]

**Comment by thoov** [comment]
Will need to have: https://github.com/embroider-build/embroider/pull/742 landed and released

**Comment by thoov** [comment]
@rwjblue this is good to go now

**Comment by rwjblue** [comment]
🎉🥁🙌👏

File: blueprints/app/files/package.json [hunk 1]
```diff
@@ -25,9 +25,9 @@
   "devDependencies": {
     "@ember/optional-features": "^2.0.0",
     "@ember/test-helpers": "^2.2.0<% if (embroider) { %>",
-    "@embroider/compat": "^0.37.0",
-    "@embroider/core": "^0.37.0",
-    "@embroider/webpack": "^0.37.0<% } %>",
+    "@embroider/compat": "^0.39.1",
```
[the remark this answers is not in the document]
Comment (rwjblue) [reply blueprints/app/files/package.json:28]: Not necessarily (we have other non-1.0.0 things in the blueprint, e.g. `ember-maybe-import-regenerator`), but we absolutely are working towards a 1.0.0. Look out for a meta issue on Embroider repo soon for that. I'd like to get that landed by the time we do 3.28.0 (an LTS release).

[the remark this answers is not in the document]
Comment (rwjblue) [reply blueprints/app/files/package.json:28]: Cross linking the issue you made to track that also: https://github.com/embroider-build/embroider/issues/761

**Comment by simonihmig** [comment]
As much as I am excited about Embroider (having used and contributed to it), I am nevertheless *quite* surprised by this move here!

Given how much this changes the underlying build infrastructure, wouldn't an RFC be required here? Mainly for two reasons:
* gather concerns about stability and DX. When this lands in a stable release, any bugs or problems Embroider has (which are many actually, at least by looking at open issues) will be contributed to "Ember". 
* draw the whole picture of what needs to be done to fully land this from the user's perspective. This especially refers to documentation, which a RFC would normally have to go into details

Some examples for these points off the top of my head:
* addons that use `postprocessTree` (for type `all`) don't work. This still affects popular ones like https://github.com/ef4/prember/issues/59, https://github.com/ember-intl/ember-intl/issues/1544, https://github.com/DockYard/ember-service-worker/issues/187, `broccoli-asset-rev`. Probably many more. That `postprocessTree` API is intentionally not supported by Embroider, but the DX ist still quite bad when users install addons from emberobserver's Top 100 list and they just don't work, without clear feedback.
* https://guides.emberjs.com/release/addons-and-dependencies/ specifically mentions `ember-cli-sass`, but this addon still has major issues with Embroider (https://github.com/embroider-build/embroider/issues/138)
* https://cli.emberjs.com/release/advanced-use/asset-compilation/#fingerprintingandcdnurls goes into details about fingerprinting, none of which work in Embroider (at least using `broccoli-asset-rev`, though Embroider/Webpack comes with some limited fingerprinting ootb, but just the main JS/CSS files IIRC)
* Embroider specific settings like route-based splitting would need to be documented in the official guides I think

Tl;dr: I feel this is premature and would require a thorough transition path in an RFC! 🤷‍♂️

/cc @ef4
